ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4086,6 +4091,37 @@ public List<String> getPartitionNames(String dbName, String tblName,
     return names;
   }
 
+  // get partition names from provided partition values
+  public List<String> getPartitionNames(String dbName, String tblName,
```

Comment:
it seems to be a partial code duplication. why not extract and reuse same logic from `List<String> getPartitionNames(String dbName, String tblName, Map<String, String> partSpec, short max)`

Reply:
`max` should be the last param

Reply:
should it be public?

Reply:
1. fixed in line 4069

2. setting `max` to the last param results in ambiguous method calls when public List<String> `getPartitionNames(String dbName, String tblName, Map<String, String> partSpec, short max)` is called with `partSpec` set to null.

3. Third party users using HS2 can benefit from this in order to get partition names with just given filter values.